scipy/stats/__init__.py

```diff
@@ -265,76 +266,165 @@
+the wide variety of statistical procedures, any attempt at coarse-grained
+categorization will be imperfect. Also, note that tests within the same heading
+are not interchangeable in general (e.g. many have different distributional
+assumptions).
+
+One Sample Tests / Paired Sample Tests
+--------------------------------------
+One sample tests are typically used to assess whether a single sample was
+drawn from a specified distribution or a distribution with specified properties
+(e.g. zero mean).
 
 .. autosummary::
    :toctree: generated/
 
-   f_oneway
-   alexandergovern
+   ttest_1samp
+   binomtest
+   skewtest
+   kurtosistest
+   normaltest
+   jarque_bera
+   shapiro
+   anderson
+   cramervonmises
+   ks_1samp
+   goodness_of_fit
+   chisquare
+   power_divergence
+
+Paired sample tests are often used to assess whether two samples were drawn
+from the same distribution; they differ from the independent sample tests below
+in that each observation in one sample is treated as paired with a
+closely-related observation in the other sample (e.g. when environmental
+factors are controlled between observations within a pair but not among pairs).
+They can also be interpreted or used as one-sample tests (e.g. tests on the
+mean or median of *differences* between paired observations).
+
+.. autosummary::
+   :toctree: generated/
+
+   ttest_rel
+   wilcoxon
+
+Association/Correlation Tests
+-----------------------------
+
+These tests are often used to assess whether there is a relationship (e.g.
+linear) between paired observations in multiple samples or among the
+coordinates of multivariate observations.
+
+.. autosummary::
+   :toctree: generated/
+
+   linregress
    pearsonr
    spearmanr
    pointbiserialr
    kendalltau
    weightedtau
    somersd
-   linregress
    siegelslopes
    theilslopes
+   page_trend_test
    multiscale_graphcorr
 
-Statistical tests
-=================
+These association tests and are to work with samples in the form of contingency
+tables. Supporting functions are available in `scipy.stats.contingency`.
+
+.. autosummary::
+   :toctree: generated/
+
+   chi2_contingency
+   fisher_exact
+   barnard_exact
+   boschloo_exact
+
+Independent Sample Tests
+------------------------
+Independent sample tests are typically used to assess whether multiple samples
+were independently drawn from the same distribution or different distributions
+with a shared property (e.g. equal means).
+
+Some tests are specifically for comparing two samples.
 
 .. autosummary::
    :toctree: generated/
 
-   ttest_1samp
-   ttest_ind
    ttest_ind_from_stats
-   ttest_rel
-   chisquare
-   cramervonmises
+   poisson_means_test
+   ttest_ind
+   mannwhitneyu
+   brunnermunzel
+   mood
+   ansari
    cramervonmises_2samp
-   power_divergence
-   kstest
-   ks_1samp
-   ks_2samp
    epps_singleton_2samp
-   mannwhitneyu
-   tiecorrect
-   rankdata
-   ranksums
-   wilcoxon
+   ks_2samp
+   kstest
```

Comment:
I'm not finding a place for `kstest` in this structure, and I think the root cause is discussed in gh-9485.